src/http_api/server.cpp

```diff
@@ -29,19 +29,166 @@ extern ConfigManager g_config;
 namespace http_api
 {
 
+using IpEndpoint = asio::ip::tcp::endpoint;
+using IpAcceptor = asio::ip::tcp::acceptor;
+using LocalEndpoint = asio::local::stream_protocol::endpoint;
+using LocalAcceptor = asio::local::stream_protocol::acceptor;
+
+namespace detail
+{
+
+template<typename Acceptor, typename Endpoint>
+void open(Acceptor& acceptor, Endpoint endpoint, ErrorCode& error)
+{
+	acceptor.open(endpoint.protocol(), error);
+	if (error) {
+		return;
+	}
+
+	asio::socket_base::reuse_address option(true);
+	acceptor.set_option(option, error);
+	if (error) {
+		return;
+	}
+
+	acceptor.bind(endpoint, error);
+	if (error) {
+		return;
+	}
+
+	acceptor.listen(asio::socket_base::max_connections, error);
+	if (error) {
+		return;
+	}
+}
+
+} //namespace detail
+
+const static std::string LOCAL_SOCKET_PREFIX {"unix://"};
+
+class GenericStreamAcceptor
+{
+public:
+	using Pointer = std::unique_ptr<GenericStreamAcceptor>;
+	using AcceptHandler = std::function<void(ErrorCode)>;
+
+	static Pointer make(IoService& service, const std::string& address, uint16_t port);
+
+	virtual ~GenericStreamAcceptor() = default;
+
+	virtual void open(const std::string& address, uint16_t port, ErrorCode& error) = 0;
+	virtual void close(ErrorCode& error) = 0;
+	virtual void asyncAccept(Peer::Socket& socket, AcceptHandler handler) = 0;
+};
+
+class LocalAcceptorWrapper : public GenericStreamAcceptor
+{
+	LocalAcceptor acceptor;
+
+	void open(const std::string& address, uint16_t port, ErrorCode& error) override;
+	void close(ErrorCode& error) override;
+	void asyncAccept(Peer::Socket& socket, AcceptHandler handler) override;
+
+public:
+	LocalAcceptorWrapper(IoService& service);
+};
```

Comment:
I didn't have time to clean it up yet. This is the 4th approach to GenericStreamAcceptor :D (I've tried using a union, boost:variant and boost::any) but all of them turned out to be incredibly messy. I plan to move the acceptor to a separate cpp file to hide all the gory details of platform dependent #ifdef-ology.